runtime/vm/jnimisc.cpp

```diff
@@ -829,7 +829,12 @@ getStringUTFLength(JNIEnv *env, jstring string)
 
 	UDATA utfLength = getStringUTF8Length(currentThread, stringObject);
 	VM_VMAccess::inlineExitVMToJNI(currentThread);
-	return (jsize)utfLength;
+
+	jsize result = INT_MAX;
+	if (utfLength <= INT_MAX) {
+		result = (jsize)utfLength;
```

Comment:
The type of `utfLength` should be `IDATA`; see `vm_api.h`.

The upper bound should be `INT32_MAX` instead of `INT_MAX`.

There are many other places that use the wrong type and are missing the check being proposed here.